uropa/visualization.py

```diff
@@ -74,30 +126,76 @@ def peak_count_plot(table, var, kind):
     pass
 
 
-def upset_plot(table, var):
+def upset_plot(table, var="feature", peak_columns=["peak_chr", "peak_start", "peak_end", "peak_strand"], title=None, title_size=20, path=None, dpi=300.0, **kwargs):
     """
     Visualize overlaps in a set with an upset plot.
     Similar to a Venn diagram but more readable especially with higher category count.
-
     For more information see: https://ieeexplore.ieee.org/document/6876017
-
     https://upsetplot.readthedocs.io/en/stable/
-
+    
     Parameters
     ----------
     table : pd.DataFrame
         Pandas dataframe containing the data.
-    var : <datatype>, <default value>
-        <param description>
-    TODO add more parameters
-
+    var : String, default="feature"
+        Column which's values to list per peak and plot as categories.
+    peak_columns: List of Strings, default=["peak_chr", "peak_start", "peak_end", "peak_strand"]
+        Colums in table which identify peaks, to be used in group_by function.
+    title : String, default=None
+        Value for title of plot. If None plot has no title.
+    title_size: Integer, default=20
+        Value for size in points of figure title.
+    path : String, default=None
+        Value with path to save plot at including plot name and file type ending. If None plot is not saved.
+    dpi : Float, default=300.0
+        Value with DPI to save plot with. If default 300.0 DPI is set.
+    **kwargs : Additional arguments
+        Any additional arguments will be passed on to the upsetplot.plot() function. For a list of possible arguments check
+        here: https://upsetplot.readthedocs.io/en/stable/api.html#upsetplot.UpSet
+    
     Returns
     -------
-    <datatype> :
-        <return description>
-    TODO should return the plotting object
+    matplotlib.figure.Figure fig :
+        Returns the plotting object.
     """
-    pass
+    
+    # Check if parameter var is valid column name
+    if var not in table.columns:
+        raise ValueError("Incorrect var parameter. Please choose a valid column name to group by.")
+    
+    # Check if values for peak_columns are valid column names
+    for column in peak_columns:
+        if column not in table.columns:
+            raise ValueError("Incorrect peak_columns parameter. Please choose valid column names to identify peaks by.")
+        
+    # Convert var column to String so that NaN values can be joined with other values in column
+    table[var] = table[var].astype(str)
+    
+    # Creat new data frame that groupes by peaks and lists values for selected colum per peak
+    grouped_table = table.groupby(peak_columns, as_index=False)[var].apply(' '.join)

```

Comment:
This won't work if the var column contains whitespaces

Reply:
Fixed by using ',' instead of ' '. I initially tried to keep peak_columns as index but the problem is that the resulting table no longer has any other column titles (specifically the var column title) which means I have trouble working with it. Resetting the index or keeping a separate index was the only solution I could find through googling. And I don't really strictly need the peak_columns as an index going forward.

Reply:
Sry but that doesn't fix the problem. Now it won't work if a cell contains a `,`. Your function should work regardless of the contents of the cells.